network/crl.go

```diff
@@ -120,19 +149,124 @@ type CRLVerifier interface {
 type DefaultCRLVerifier struct {
 	Config CRLConfig
 	Client CRLClient
-	// key = URI of cached CRL
-	// value = parsed CRL
-	// XXX maybe hide it behind mutex?
-	cache map[string]*pkix.CertificateList
+	Cache  CRLCache
+}
+
+func (v DefaultCRLVerifier) getCachedOrFetch(uri string) ([]byte, error) {
+	crl, err := v.Cache.Get(v.Config.uri)
+	if crl != nil {
+		if err != nil {
+			return nil, err
+		} else {
+			return crl, nil
+		}
+	}
```

Comment:
This looks a bit too complex and does not handle the case where the cache itself fails. If `err != nil` then it is expected for `crl == nil`, and that case is not really handled.

I guess if we allow the cache to fail *and* want to handle this failure, `Get()` method should return `([]byte, bool, error)` which is handled like this:

```go
crl, ok, err := v.Cache.Get(v.Config.uri)
if err != nil {
	return nil, err
}
if ok {
	return crl, nil
}
// Else: no cache hit, fetch from network.
```
I don't really see why we'd care for the error here: if the cache fails, we go straight to querying the origin anyway. So I guess the `CRLCache` interface can be changed to this:

```go
type CRLCache interface {
	Get(key string) ([]byte, bool)     // true if cached, false if absent
	Put(key string, value []byte) bool // true if replaced, false if first insert
	Remove(key string) bool            // true if removed, false if never existed
}
```
I'd say that the cache is most likely going to be local, in-memory one anyway, so it does not have many failure modes that are important here.

Then the user code gets even more simple:
```go
crl, ok := v.Cache.Get(v.Config.uri)
if ok {
	return crl, nil
}
```